dali/dfuplus/dfuplus.cpp

```diff
@@ -1641,20 +1667,46 @@ int CDfuPlusHelper::erasehistory()
 
     if (backup)
     {
-        // The generated getXmlmap() return with a const Memory buffer reference
-        const MemoryBuffer &respMsg = resp->getXmlmap();
-        if (0 == respMsg.length())
+        IArrayOf<IConstHistory>& arrHistory = resp->getFileHistory();
+        if (0 == arrHistory.length())
         {
             error("%s doesn't have stored history!\n", lfn);
             return -2;
         }
 
-        Owned<IPropertyTree> history;
-        // Based on the missing createPTree( const MemoryBuffer &)
-        // need to cast from const MemeoryBuffer & to MemoryBuffer & to create PTree
-        history.setown(createPTree((MemoryBuffer &)respMsg));
+        StringBuffer out;
+        out.append("<History>").newline();
+        for(unsigned i = 0; i < arrHistory.length(); i++)
+        {
+            IConstHistory* historyRecord = &arrHistory.item(i);
+            if(historyRecord != NULL)
+            {
+                out.appendf("\t<Origin ip=\"%s\"", historyRecord->getIP()).newline();
```

Comment:
why do we use tabs for xml but 3 spaces for JSON, is that part of a standard? if not we might want to stay consistent.